kotlin/services/appsync/src/main/kotlin/com/example/appsync/CreateDataSource.kt

```diff
@@ -0,0 +1,80 @@
+// snippet-sourcedescription:[CreateDataSource.kt demonstrates how to create an AWS AppSync data source that uses Amazon DynamoDB.]
+// snippet-keyword:[AWS SDK for Kotlin]
+// snippet-service:[AWS AppSync]
+// snippet-keyword:[Code Sample]
+// snippet-sourcetype:[full-example]
+// snippet-sourcedate:[04-01-2022]
+// snippet-sourceauthor:[scmacdon - AWS]
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.appsync
+
+// snippet-start:[appsync.kotlin.create_ds.import]
+import aws.sdk.kotlin.services.appsync.AppSyncClient
+import aws.sdk.kotlin.services.appsync.model.CreateDataSourceRequest
+import aws.sdk.kotlin.services.appsync.model.DataSourceType
+import aws.sdk.kotlin.services.appsync.model.DynamodbDataSourceConfig
+import kotlin.system.exitProcess
+// snippet-end:[appsync.kotlin.create_ds.import]
+
+/**
+ * Before running this Kotlin code example, set up your development environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
+ */
+
+suspend fun main(args: Array<String>) {
+
+    val usage = """
+        Usage:
+            <apiId> <dsName> <dsRole> <tableName>
+
+        Where:
+            apiId - The id of the API (you can obtain the value from the AWS Management console). 
+            dsName - The name of the data source. 
+            dsRole - The Identity and Access Management (IAM) service role for the data source. 

```

Comment:
Identity and Access Management (IAM) --> AWS Identity and Access Management (IAM)

Reply:
fixed